File: edgeneoapp/src/app/service/i-service.ts
import { Observable } from "rxjs";
import { RespostaPaginada } from "../model/resposta-paginada";

export interface IService<T> {
    apiUrl: string;
    get(termoBusca?: string): Observable<T[]>;
    getById(id: number): Observable<T>;
    save(objeto: T): Observable<T>;
    delete(id: number): Observable<void>;
}

File: edgeneoapi/src/main/java/br/ufac/edgeneoapi/service/IService.java
package br.ufac.edgeneoapi.service;

import java.util.List;

public interface IService<T> {
    List<T> getAll();
    T getById(Long id);
    // List<T> busca(String termoBusca);
    T save(T objeto);
    void delete(Long id);
}